### PR TITLE
box plots data scripts

### PR DESCRIPTION
**Pull request by msvinaykumar**
# Add tests for box plots

## Description

This PR will do the following changes - 
- add new tests for the box plots feature following the scenarios mentioned in  #1116 

### Type of change

- [ ] Refactoring Code
- [ ] New feature
- [x] Test changes

## How has this been tested?

- Using kruize-demo scripts. 
Image: `quay.io/khansaad/autotune_operator:boxplots_fixes`
- Manually verified the plots data.

**Test Configuration**
* Kubernetes clusters tested on: minikube

## Checklist :dart:

- [X] Followed coding guidelines
- [X] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [x] Tests added or updated

## Additional information
-

### CODE REVIEW

File: tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py [hunk 1]
```diff
@@ -5,6 +5,132 @@
 from helpers.utils import *
 
 
+@pytest.mark.sanity
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py:8]: @msvinaykumar - You can use the term tests and other existing tests itself to validate the plots, update the json schemas and utils function to validate recommendations

Comment (msvinaykumar) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py:8]: It appears that the TermTest is utilizing the validate_container function, which means that whatever TermTest validates, plots as well

**Comment by khansaad** [comment]
PR check is failing because of the newly added box plot validations test. This should pass once PR #1189 is merged.

Also, have added the remaining test for validation. Running the regression on jenkins:
https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/Autotune/job/kruize_test_pull_request/96/

cc: @msvinaykumar @chandrams

File: tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py (hunk 1 above)
Comment (chandrams) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py:8]: @khansaad - json schemas need updation to validate plots

File: tests/scripts/remote_monitoring_tests/helpers/utils.py [hunk 2]
```diff
@@ -557,13 +561,18 @@ def validate_container(update_results_container, update_results_json, list_reco_
...
-                        if None != recommendation_engines_object:
+                        if recommendation_engines_object is not None:
                             for engine_entry in engines_list:
                                 if engine_entry in terms_obj[term]["recommendation_engines"]:
                                     engine_obj = terms_obj[term]["recommendation_engines"][engine_entry]
                                     validate_config(engine_obj["config"], metrics)
                                     validate_variation(current_config, engine_obj["config"], engine_obj["variation"])
-
+                        # validate Plots data
+                        if PLOTS in terms_obj[term]:
```
[the remark this answers is not in the document]
Comment (khansaad) [reply tests/scripts/remote_monitoring_tests/helpers/utils.py:571]: This method is being called during the listRecommendations basic validations stage as well before the call to updateRecommendations so at that time `plots` data won't be available.

File: manifests/crc/default-db-included-installation/minikube/kruize-crc-minikube.yaml [hunk 3]
```diff
@@ -110,7 +110,7 @@ data:
       "monitoringendpoint": "prometheus-k8s",
       "savetodb": "true",
       "dbdriver": "jdbc:postgresql://",
-      "plots": "false",
+      "plots": "true",
```
[the remark this answers is not in the document]
Comment (khansaad) [reply manifests/crc/default-db-included-installation/minikube/kruize-crc-minikube.yaml:113]: Updated

File: tests/scripts/remote_monitoring_tests/helpers/utils.py [hunk 4]
```diff
@@ -514,8 +518,8 @@ def validate_container(update_results_container, update_results_json, list_reco_
                 terms_obj = list_reco_container["recommendations"]["data"][interval_end_time]["recommendation_terms"]
                 current_config = list_reco_container["recommendations"]["data"][interval_end_time]["current"]
 
-                duration_terms = ["short_term", "medium_term", "long_term"]
-                for term in duration_terms:
+                duration_terms = {'short_term': 4, 'medium_term': 7, 'long_term': 15}
```
[the remark this answers is not in the document]
Comment (khansaad) [reply tests/scripts/remote_monitoring_tests/helpers/utils.py:521]: Updated

File: tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py (hunk 1 above)
Comment (khansaad) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py]: Updated all the schemas

**Comment by khansaad** [comment]
> PR check is failing because of the newly added box plot validations test. This should pass once PR #1189 is merged.
> 
> Also, have added the remaining test for validation. Running the regression on jenkins: https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/Autotune/job/kruize_test_pull_request/96/
> 
> cc: @msvinaykumar @chandrams

This one ran successfully. Re-running the whole suite after schema updates:
https://ci.app-svc-perf.corp.redhat.com/job/ExternalTeams/job/Autotune/job/kruize_test_pull_request/97/

File: tests/scripts/remote_monitoring_tests/helpers/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (chandrams) [reply tests/scripts/remote_monitoring_tests/helpers/utils.py:571]: We need to find another way to address that. Adding a condition to check if plots if present might not catch a bug when really plots is missing in the recommendations

[the remark this answers is not in the document]
Comment (khansaad) [reply tests/scripts/remote_monitoring_tests/helpers/utils.py]: Added a flag now in all the cases where updateRecommendations call is made. We'll check for plots whenever updateRecommendations is made.

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py [hunk 5]
```diff
@@ -1054,7 +1070,9 @@ def test_list_recommendations_for_diff_reco_terms_with_only_latest(test_name, nu
         exp_found = False
         for list_reco in list_reco_json:
             if create_exp_json[0]['experiment_name'] == list_reco['experiment_name']:
-                validate_reco_json(create_exp_json[0], update_results_json, list_reco, expected_duration_in_hours, test_name)
+                plots_available = False
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py:1073]: @khansaad Setting this flag will not validate the terms that have recommendations.

**Comment by dinogun** [comment]
@khansaad Please fix the conflicts

File: tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py (hunk 5 above)
Comment (khansaad) [reply tests/scripts/remote_monitoring_tests/rest_apis/test_list_recommendations.py]: Removed the flag, able to run the tests successfully.

**Pull request merged**